js/modules/timeline.src.js

```diff
@@ -41,11 +41,28 @@ wrap(H.SVGElement.prototype, 'animate', function (proceed, params) {
     }
     return proceed.apply(this, Array.prototype.slice.call(arguments, 1));
 });
-
 /**
- * Timeline Series definition.
+ * The timeline series type.
+ *
+ * @constructor seriesTypes.timeline
+ * @augments Series
  */
 
+/**
+ * The timeline series presents given events along a drawn line.
+ *
+ * @sample highcharts/series-timeline/alternate-labels Timeline series
+ *
+ * @extends plotOptions.line
+ * @excluding animationLimit,boostThreshold,connectEnds,connectNulls,
+ *          cropThreshold,dashStyle,findNearestPointBy,getExtremesFromAll,
+ *          lineWidth,marker,negativeColor,pointInterval,pointIntervalUnit,
+ *          pointPlacement,pointStart,softThreshold,stacking,step,threshold,
+ *          turboThreshold,zoneAxis,zones
+ * @product highcharts
+ * @since 7.0.0
+ * @optionparent plotOptions.timeline
+ */
 seriesType('timeline', 'line', {
     showInLegend: false,
```

Comment:
What is the problem with support of legend?

Reply:
The problem was, that all points from the series had been present in the legend as one series. Now every event is a separate legend item.